File: Data Structures & Algorithms/trapping-rain-water/submission-0.java
class Solution {
    private int findRightMaximum(int st, int [] height){
        int res=-1;
        for(int i=st; i<height.length; i++){
          res = Math.max(res,height[i]);
        }
        return res;
    }
    public int trap(int[] height) {
        int totalWater=0;
        int leftMax=height[0];
        for(int i=1; i<height.length; i++){
           
           int rightMax= findRightMaximum(i+1,height);
           int cur=height[i];
           
           if(leftMax > cur && rightMax > cur){
              int curWater = Math.min(leftMax,rightMax) - cur;
              totalWater += curWater;
           }
           
           leftMax = cur > leftMax ? cur : leftMax;

        }

        return totalWater;
        
    }
}

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-0.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        var mark = new int[26];
        int st=0;
        int maxFreq=0,maxLength=0;
        for(int end = 0; end <s.Length; end++){
            char ch = s[end];
            mark[ch-'A']++;
            maxFreq = Math.Max(maxFreq,mark[ch-'A']);

            while((end-st+1)-maxFreq > k){
               mark[s[st]-'A']--;
               st++;
            }
           maxLength = Math.Max(maxLength,end-st+1);
        }
        return maxLength;
    }
}
